src/emissions_harmonization_historical/region_mapping.py

```diff
@@ -3,11 +3,36 @@
 from pathlib import Path
 
 import pandas as pd
+import requests
 from nomenclature import countries
 from nomenclature.definition import DataStructureDefinition
 from nomenclature.processor import RegionProcessor
 
 
+def get_latest_commit_hash(repo_owner, repo_name, fallback_commit="cc69ed0a415a63c7ce7372d5a36c088d9cbee055"):
+    """
+    Fetch the latest commit hash from a GitHub repository.
+
+    Falls back to a default hash, here for IAMconsortium/common-definitions, if the request fails.
+
+    Args:
+        repo_owner (str): GitHub user.
+        repo_name (str): Repository name.
+        fallback_commit (str): Commit hash to use if request fails.
+
+    Returns
+    -------
+        str: The latest commit hash or the fallback hash.
+    """
```

Comment:
Minor thing, but you're blending numpy style docstrings and google style docstrings. In everything I do, I use numpy (but honestly, both are common so just pick your favourite and use it)

Here's the numpy equivalent. You can move the type hints to the function definition to make everything a bit simpler

```suggestion
def get_latest_commit_hash(repo_owner: str, repo_name: str, fallback_commit: str="cc69ed0a415a63c7ce7372d5a36c088d9cbee055"):
    """
    Fetch the latest commit hash from a GitHub repository.

    Falls back to a default hash, here for IAMconsortium/common-definitions, if the request fails.

    Parameters
    -----------
    repo_owner
        GitHub user
        
    repo_name
        Repository name
    
    fallback_commit
        Commit hash to use if request fails.

    Returns
    -------
    :
        The latest commit hash or the fallback hash.
    """
```